libnavigation-core/src/main/java/com/mapbox/navigation/core/telemetry/MapboxNavigationTelemetry.kt

```diff
@@ -343,73 +458,120 @@ internal object MapboxNavigationTelemetry {
                 }
                 onEventUpdated?.let { it(feedbackEvent) }
             }
-        } else {
-            logger?.e(
-                TAG,
-                Message(
-                    "User Feedback event creation failed. The event can only be created in " +
-                        "active guidance (trips session started and route is available)."
-                )
-            )
         }
     }
 
-    fun unregisterListeners(mapboxNavigation: MapboxNavigation) {
-        mapboxNavigation.run {
-            unregisterLocationObserver(locationsCollector)
-            unregisterRouteProgressObserver(routeProgressObserver)
-            unregisterRoutesObserver(routesObserver)
-            unregisterOffRouteObserver(offRouteObserver)
-            unregisterNavigationSessionObserver(navigationSessionStateObserver)
-            unregisterArrivalObserver(arrivalObserver)
+    private fun processDeparture() {
+        if (!routeData.hasRouteAndRouteProgress()) {
+            log("cannot handle depart: $LOG_TELEMETRY_NO_ROUTE_OR_ROUTE_PROGRESS")
+            return
         }
-        MapboxMetricsReporter.disable()
+        sendMetricEvent(
+            NavigationDepartEvent(PhoneState(applicationContext)).apply { populate() }
+        )
     }
 
-    private fun handleStateChanged(
+    private fun getFreeDriveEvent(
         oldState: NavigationSession.State,
         newState: NavigationSession.State
-    ) {
-        when {
-            oldState == FREE_DRIVE && newState == IDLE -> trackFreeDrive(STOP)
-            oldState == FREE_DRIVE && newState == ACTIVE_GUIDANCE -> trackFreeDrive(STOP)
-            oldState != FREE_DRIVE && newState == FREE_DRIVE -> trackFreeDrive(START)
+    ): FreeDriveEventType? {
+        return when {
+            oldState == FREE_DRIVE && newState == IDLE -> STOP
+            oldState == FREE_DRIVE && newState == ACTIVE_GUIDANCE -> STOP
+            oldState != FREE_DRIVE && newState == FREE_DRIVE -> START
+            else -> null
         }
     }
 
-    private fun trackFreeDrive(type: FreeDriveEventType) {
-        log("trackFreeDrive $type")
-        dynamicFreeDriveValues.run {
-            if (type == START) {
-                sessionId = obtainUniversalUniqueIdentifier()
-                sessionStartTime = Date()
+    private fun handleTelemetryState() {
+        val localTelemetryState = telemetryState
+        when (sessionState) {
+            IDLE -> {
+                if (localTelemetryState is NavTelemetryState.Running) {
+                    telemetryState = NavTelemetryState.Paused(
+                        SessionMetadataOnPause(
+                            navigatorSessionIdentifier = navObtainUniversalUniqueIdentifier(),
+                        )
+                    )
```

Comment:
What is the purpose of the paused state? I see that the next active guidance or free drive session will inherit the generated here identifier, but why? Couldn't they generate their own? The final result would be exactly the same.

Reply:
`Paused` means telemetry is started, but `SessionState` is `Idle`.
Between `Start` and `Stop` _telemetry_ it has the same `navigatorSessionIdentifier`, even if the `session` is _stopped_(`MapboxNavigation#stopTripSession`), telemetry hold its own session. The same approach is running with the iOS SDK.
Via the same `navigatorSessionIdentifier` we can concatenate different combinations of `freeDrive`-`activeGuidance`, which are run under the same session